source/opt/instruction.h

```diff
@@ -239,7 +241,45 @@ class Instruction : public utils::IntrusiveNodeBase<Instruction> {
   void ReplaceOperands(const std::vector<Operand>& new_operands);
 
   // Returns true if the instruction annotates an id with a decoration.
-  inline bool IsDecoration();
+  inline bool IsDecoration() const;
+
+  // Returns true if the instruction is known to be a load from read-only
+  // memory.
```

Comment:
Document why |ctx| is required (in the others too).

Reply:
I'll skip that.  If #969 goes through first, I'll be removing the parameter, and just using the context in the instruction if it exists.

Reply:
Sounds good.